### ISSUE TITLE
Feature Request: Support for Models with Nested Estimators

### ISSUE THREAD

**Issue by Gnpd** [SF-Tec/openmodels#25]
### Description

Currently, the `SklearnSerializer` does not fully support serialization and deserialization of scikit-learn models that contain **nested estimators** (e.g., meta-estimators such as `Pipeline`, `RandomForestRegressor`, etc.). These models often have attributes or parameters that are themselves estimators or lists/dictionaries of estimators, which are not handled recursively by the current serialization logic.

### Motivation

Many advanced machine learning workflows in scikit-learn rely on meta-estimators that wrap or compose other estimators. Being able to serialize and deserialize these composite models is essential for:
- Saving and loading complex model pipelines
- Model deployment and reproducibility
- Interoperability with other tools and formats

### Current Behavior

- The serializer only serializes the top-level estimator’s attributes.
- If an attribute is itself an estimator (or a list/dict of estimators), it is not recursively serialized, leading to loss of information or errors during deserialization.
- Some meta-estimators are explicitly listed as unsupported in `NOT_SUPPORTED_ESTIMATORS` due to this limitation.

### Desired Behavior

- The serializer should **recursively serialize** any nested estimators found in attributes (including lists, tuples, and dictionaries of estimators).
- During deserialization, the serializer should be able to **reconstruct the full nested structure** of the original model, including all sub-estimators and their parameters.

### Example

```python
from sklearn.pipeline import Pipeline
from sklearn.preprocessing import StandardScaler
from sklearn.linear_model import LogisticRegression

pipe = Pipeline([
    ('scaler', StandardScaler()),
    ('clf', LogisticRegression())
])
pipe.fit(X, y)

serializer = SklearnSerializer()
serialized = serializer.serialize(pipe)
restored = serializer.deserialize(serialized)
assert type(restored) is Pipeline
```

### Additional Notes

- Care should be taken to handle circular references and shared estimators.
- Consider updating the `NOT_SUPPORTED_ESTIMATORS` list as support is added.

**Comment by Gnpd** [comment]
Ready on PR #28
